massa-db-worker/src/massa_db.rs

```diff
@@ -837,3 +850,803 @@ impl MassaDBController for RawMassaDB<Slot, SlotSerializer, SlotDeserializer> {
+        ));
+
+        // Add data 1 (at slot 1)
+        let batch_key_1 = vec![1, 2, 3];
+        let batch_value_1 = vec![4, 5, 6];
+        let batch_1 = DBBatch::from([(batch_key_1.clone(), Some(batch_value_1.clone()))]);
+        let versioning_batch_1 =
+            DBBatch::from([(vec![10, 20, 30], Some(vec![127, 128, 254, 255]))]);
+        let slot_1 = Slot::new(1, 0);
+        let mut guard = db.write();
+        guard.write_batch(batch_1.clone(), versioning_batch_1.clone(), Some(slot_1));
+        drop(guard);
+
+        // Add data 2 (at slot 1)
+        let batch_key_2 = vec![11, 22, 33];
+        let batch_value_2 = vec![44, 55, 66];
+        let batch_2 = DBBatch::from([(batch_key_2.clone(), Some(batch_value_2.clone()))]);
+        let versioning_batch_2 = DBBatch::from([(vec![12, 23, 34], Some(vec![255, 254, 253]))]);
+        let slot_2 = Slot::new(2, 0);
+        let mut guard = db.write();
+        guard.write_batch(batch_2.clone(), versioning_batch_2.clone(), Some(slot_1));
+        drop(guard);
+
+        // Stream using StreamingStep::Started
+        let last_state_step: StreamingStep<Vec<u8>> = StreamingStep::Started;
+        let stream_batch_ = db.read().get_batch_to_stream(&last_state_step, None);
+        let stream_batch = stream_batch_.unwrap();
+        assert_eq!(
+            stream_batch.new_elements,
+            BTreeMap::from([
+                (batch_key_1.clone(), batch_value_1.clone()),
+                (batch_key_2.clone(), batch_value_2.clone())
+            ])
+        );
+        assert_eq!(stream_batch.updates_on_previous_elements, BTreeMap::new());
+        assert_eq!(stream_batch.change_id, slot_1);
+
+        // Now updates some value (at slot 2)
+        let batch_value_2b = vec![94, 95, 96];
+        let batch_2b = DBBatch::from([(batch_key_2.clone(), Some(batch_value_2b.clone()))]);
+        let versioning_batch_2b = DBBatch::new();
+        let mut guard = db.write();
+        guard.write_batch(batch_2b.clone(), versioning_batch_2b.clone(), Some(slot_2));
+        drop(guard);
+
+        // Add data (at slot 3)
+        let batch_key_3 = vec![61, 62, 63];
+        let batch_value_3 = vec![4, 5, 6, 7];
+        let batch_3 = DBBatch::from([(batch_key_3.clone(), Some(batch_value_3.clone()))]);
+        let versioning_batch_3 = DBBatch::new();
+        let slot_3 = Slot::new(3, 0);
+        let mut guard = db.write();
+        guard.write_batch(batch_3.clone(), versioning_batch_3.clone(), Some(slot_3));
+        drop(guard);
+
+        // Stream using StreamingStep::Finished
+
+        let last_state_step: StreamingStep<Vec<u8>> =
+            StreamingStep::Finished(Some(batch_key_2.clone()));
+        let stream_batch_ = db
+            .read()
+            .get_batch_to_stream(&last_state_step, Some(slot_1));
+        let stream_batch = stream_batch_.unwrap();
+
+        // FIXME?: new_elements is empty, everything is on updates_on_previous_elements
```

Comment:
Ah, I think we should add a comment somewhere (in lib.rs or in StreamBatch definition), indeed here "new" keys will be streamed in the updates field.
Updates represent "everything that changed since the last change_id streamed, up to a certain key".
This definition also applies to keys that were not in the DB beforehand.